map/query-features-with-arcade-expression/src/main/java/com/esri/samples/query_features_with_arcade_expression/QueryFeaturesWithArcadeExpressionSample.java

```diff
@@ -0,0 +1,197 @@
+                Point mapPoint = mapView.screenToLocation(screenPoint);
+
+                var identifyLayerResultFuture = mapView.identifyLayerAsync(policeBeatsLayer,
+                  screenPoint, 0, false);
+                identifyLayerResultFuture.addDoneListener(() -> {
+                  if (identifyLayerResultFuture.isDone()) {
+                    try {
+                      var identifyLayerResult = identifyLayerResultFuture.get();
+
+                      // only execute query if the police beats layer has data at the selected point
+                      if (identifyLayerResult.getElements().size() != 0) {
+                        ArcGISFeature feature = (ArcGISFeature) identifyLayerResult.getElements().get(0);
+
+                        // show callout and text without data
+                        mapView.getCallout().setTitle(calloutText);
+                        mapView.getCallout().setDetail("...");
+                        mapView.getCallout().showCalloutAt(mapPoint);
+                        mapView.getCallout().setVisible(true);
+
+                        // setup arcade expression and evaluator
+                        String arcadeExpressionString = "var crimes = FeatureSetByName($map, 'Crime in the last 60 days');\n" +
+                          "return Count(Intersects($feature, crimes));";
+                        var arcadeExpression = new ArcadeExpression(arcadeExpressionString);
+                        var arcadeEvaluator = new ArcadeEvaluator(arcadeExpression, ArcadeProfile.FORM_CALCULATION);
+
+                        // instantiate key/value pairs
+                        var hashMap = new HashMap<String, Object>();
+                        hashMap.put("$feature", feature);
+                        hashMap.put("$map", mapView.getMap());
+
+                        // evaluate the arcade expression asynchronously
+                        ListenableFuture<ArcadeEvaluationResult> resultFuture = arcadeEvaluator.evaluateAsync(hashMap);
+
+                        // wait for the expression to finish evaluating before attempting to access it
+                        resultFuture.addDoneListener(() -> {
+                          if (resultFuture.isDone()) {
+                            try {
+                              // get result from async method and cast to int
+                              var arcadeEvaluationResult = resultFuture.get();
+                              var crimesCount = Math.round((double) arcadeEvaluationResult.getResult());
+
+                              // add data from arcade evaluation to callout
+                              mapView.getCallout().setDetail(String.valueOf(crimesCount));
+                            } catch (Exception e) {
+                              throw new RuntimeException(e);
+                            }
+                          }
+                        });
+                      } else {
```

Comment:
I think we could add an additional comment in here to help the user keep track of what condition this is handling

```suggestion
                      } else {
                      // if no data is found hide the callout

```